### DISCUSSION TITLE
ZFSBootMenu and LUKS

### DISCUSSION THREAD

**Discussion by kuwairo** [zbm-dev/zfsbootmenu#513] [Questions] [answered]
Hello. Is it possible to use ZFS on LUKS with ZFSBootMenu? If so, what would the configuration look like (roughly)?

**Comment by ahesford** [comment, marked as answer]
There is no built-in support for LUKS. However, it should be possible to build a custom ZFSBootMenu image using the dracut `crypt` module or whatever the equivalent is for mkinitcpio. People reported success doing this with dracut some time ago. The relevant configuration would be in `/etc/zfsbootmenu/dracut.conf.d` or `/etc/zfsbootmenu/mkinitcpio.conf`; boot your ZFSBootMenu image with whatever kernel command-line arguments you'd need to unlock your LUKS volume in a regular distribution initramfs.

**Reply by kuwairo** [reply]
Thanks for the reply. Will try this soon then. This project is awesome :)
